Library/Homebrew/extend/os/mac/diagnostic.rb

```diff
@@ -438,6 +439,43 @@ def check_if_supported_sdk_available
           #{update_instructions}
         EOS
       end
+
+      # The CLT 10.x -> 11.x upgrade process on 10.14 contained a bug which broke the SDKs.
+      # Notably, MacOSX10.14.sdk would indirectly symlink to MacOSX10.15.sdk.
+      # This diagnostic was introduced to check for this and recommend a full reinstall.
+      def check_broken_sdks
+        locator = MacOS.sdk_locator
+
+        return if locator.all_sdks.all? do |sdk|
+          path_version = sdk.path.basename.to_s[MacOS::SDK::VERSIONED_SDK_REGEX, 1]
+          next true if path_version.blank?
+
+          path_version = MacOS::Version.new(path_version)
+
+          # Compare, excluding patch version
+          sdk.version == MacOS::Version.from_symbol(path_version.to_sym)
+        end
+
+        if locator.source == :clt
+          source = "Command Line Tools (CLT)"
+          path_to_remove = MacOS::CLT::PKG_PATH
+          installation_instructions = MacOS::CLT.installation_instructions
```

Comment:
Any chance for tests for this stuff would be great.

Reply:
By "this stuff" I mean in general the sections/new code here that are not covered by tests

Reply:
I don't quite follow what you're pointing at for tests. The code in the diagnostic check should be covered by the test now, expect the Xcode path (which I can do a test for, though it's just different text).

Reply:
Sorry, I don't even mean specifically this file, I just mean everywhere in this PR that's not been touched by tests (and gets the codecov warning) to investigate if it could be